### DISCUSSION TITLE
AttributeError: 'And' object has no attribute 'evalf'

### DISCUSSION THREAD

**Discussion by retsyo** [sympy/sympy#25628] [Q&A]
The code
```python
from sympy import *

r = Symbol('r')

s = Integer(50)/Integer(3)/(r+1)

x0 = solve(f'{s}>1')
print(f'{x0}')
```
says
```
(-1 < r) & (r < 47/3)
```
which I want to express as `(-1 < r) & (r < 15.66)`. But `x0.evalf()` says `AttributeError: 'And' object has no attribute 'evalf'`.

There are 2 questions
1. is this behaviour intented? since I think `x0.evalf()` returns `(-1 < r) & (r < 15.66)` is a nature way
2. how to get desired `(-1 < r) & (r < 15.66)` currently?

Thanks

**Comment by oscarbenjamin** [comment]
The `evalf` method should be added to `And` by subclassing `EvalfMixin`. Perhaps `Boolean` should subclass `EvalfMixin`. Perhaps `Basic` should subclass `EvalfMixin`...
